src/Aspire.Hosting/Dashboard/proto/resource_service.proto

```diff
@@ -0,0 +1,260 @@
+    optional string http_address = 4;
+}
+
+message StringArray {
+    repeated string values = 1;
+}
+
+message AdditionalData {
+    // Name of the data item, e.g. "id", "image", "path", ...
+    string name = 1;
+    // Optional namespace, e.g. "container", "executable", "project", ...
+    optional string namespace = 2;
+    // Optional display name, may be localized
+    optional string display_name = 3;
+    // The data value. May be null, a number, a string, a boolean, a dictionary of values (Struct), or a list of values (ValueList).
+    google.protobuf.Value value = 4;
+}
+
+message ResourceId {
+    string uid = 1;
+    // TODO do we need resource_type to make unique names? if not, inline ResourceId type as string.
+    string resource_type = 2;
+}
+
+// Models the full state of an resource (container, executable, project, etc) at a particular point in time.
+message ResourceSnapshot {
+    ResourceId resource_id = 1;
+    string display_name = 2;
+    optional string state = 3;
+    optional google.protobuf.Timestamp created_at = 4;
+    repeated EnvironmentVariable environment = 5;
+    optional int32 expected_endpoints_count = 6;
+    repeated Endpoint endpoints = 7;
+    repeated ResourceCommandRequest commands = 8;
+
+    // TODO potentially remove endpoints/env vars/anything that might not be general purpose
+
+    // List of additional data, as name/value pairs.
+    // For:
+    // - Containers: image, container_id, ports
+    // - Executables: process_id, executable_path, working_directory, arguments
+    // - Projects: process_id, project_path
+    repeated AdditionalData additional_data = 9;
+}
+
+////////////////////////////////////////////
+
+// Models a snapshot of resource state
+message WatchResourcesSnapshot {
+    repeated ResourceSnapshot resources = 1;
+    repeated ResourceType types = 2;
+}
+
+////////////////////////////////////////////
+
+message ResourceDeletion {
+    ResourceId resource_id = 1;
+}
+
+message WatchResourcesChange  {
+    oneof kind {
+        ResourceDeletion delete = 1;
+        ResourceSnapshot upsert = 2;
+    }
+}
+
+message WatchResourcesChanges {
+    repeated WatchResourcesChange value = 1;
+}
+
+////////////////////////////////////////////
+
+// Initiates a subscription for data about resources.
+message WatchResourcesRequest {
+    // True if the client is establishing this connection because a prior one closed unexpectedly.
+    optional bool is_reconnect = 1;
+}
+
+// A message received from the server when watching resources. Has multiple types of payload.
+message WatchResourcesUpdate {
+    oneof kind {
+        // Snapshot of current resource state. Received once upon connection, before any "changes".
+        WatchResourcesSnapshot initial_snapshot = 1;
+        // One or more deltas to apply.
+        WatchResourcesChanges changes = 2;
+    }
+}
+
+////////////////////////////////////////////
+
+message ConsoleLogLine {
+    string text = 1;
+    optional bool is_error = 2;
```

Comment:
Could you add a description for `is_error` property? I have experienced people being confused between "this is a log line that describes an application error" vs "this log line was written to standard error stream" (`stderr`, descriptor 2)

Reply:
Done.